### PR TITLE
Remove unused function html_header_show() and all usages

### LINKED ISSUE #884: Misused Function
Today, while working with bootstrap on some files ran into this function:

`<?php html_header_show(); ?>`

I checked this function in other files and it is an empty function.

Thanks,
jdew192837

### PR DESCRIPTION
**Pull request by Jdew192837**
Fixes #884

html_header_show() serves no purpose in the codebase. This Pull request thus removes it and its usages. :)

### CODE REVIEW

**Comment by Jdew192837** [comment]
@teryhill @aethelwulffe PTAL! :)

**Comment by Jdew192837** [comment]
@tmccormi PTAL as well :D

**Comment by aethelwulffe** [comment]
We got conflicts here...

**Comment by Jdew192837** [comment]
Could you, state them or are they just the conflicts from merge overlaps?

**Comment by Jdew192837** [comment]
@aethelwulffe Are they just overlapping merge conflicts

**Comment by aethelwulffe** [comment]
Just the merge conflicts.
I didn't know if you were seeing them (like checked your PR status recently) so I thought I would mention it.  My comment was terse because I am trying to look at as much stuff as I can.

**Comment by Jdew192837** [comment]
@aethelwulffe @teryhill I believe I solved them. PTAL! :)

**Comment by Jdew192837** [comment]
I think I need write access @aethelwulffe maybe or is this good? I think my commit fixed it all.

**Comment by Jdew192837** [comment]
screenshot 13

Because I went through and fixed them, it's just that I can't merge things. @aethelwulffe

**Comment by Jdew192837** [comment]
Got it, I just manually rebased it instead. :) I think this is ready for another look @aethelwulffe

**Comment by Jdew192837** [comment]
@teryhill @tmccormi @aethelwulffe 
PTAL,
Think everything is fixed, also could you take a look at #942 
Thanks,
jdew192837 (nulxis)

**Comment by Jdew192837** [comment]
@Trodrige @teryhill @tmccormi @aethelwulffe 
PTAL,
Think I got everything finished also PTAL at #942 
Thanks,
jdew192837 (nulxis)

**Comment by teryhill** [comment]
@yehster  what is your opinion on removing this?

**Comment by Jdew192837** [comment]
@aethelwulffe Do you mind approving my pr :D 
Thanks,
jdew192837 (nulxis)

**Comment by pri2si17-1997** [comment]
I also think that it should be removed and it will not break any code. But let's wait for @aethelwulffe @tmccormi @yehster review.

**Comment by yehster** [comment]
Changes like this which are to so many files always scare me a little... If the function is removed from headers.inc but a call is missed in one of the hundreds of files, there will be a fatal error...

This is another example of where we really need automated testing to verify that what we think is going to happen based on code review truly is the case.

I'm ok with removing the function as long as we are careful about it and don't treat the changes lightly.

**Comment by teryhill** [comment]
@pri2si17-1997 , @yehster  and I vote remove @tmccormi and @aethelwulffe

File: interface/forms/reviewofs/new.php [hunk 1]
```diff
@@ -13,13 +13,13 @@
 		<!-- Get Bootstrap and jQuery (required for bootstrap) -->
 		<?php call_required_libraries(['bootstrap', 'jquery-min-1-9-1']); ?>
 	    <script type="text/javascript">
-	      // Resizes the panels to be the height of the largest panel
-	      function resizePanel() {
```
Comment (aethelwulffe) [inline interface/forms/reviewofs/new.php]: Unrelated whitespace change snuck in here.
Looks like ATOM is doing its thing!

File: interface/patient_file/encounter/forms.php [hunk 2]
```diff
@@ -21,7 +21,6 @@
 <html>
 
 <head>
-<?php html_header_show();
```
Comment (aethelwulffe) [inline interface/patient_file/encounter/forms.php]: NO CLOSING TAG!  Needs OPENING TAG replaced!!!

File: interface/super/rules/base/template/basic.php [hunk 3]
```diff
@@ -7,21 +7,17 @@
...
+	</head>
 
+	<body class='body_top'>
+		<?php
+			if ( file_exists($viewBean->_view_body) ) {
+			    require_once($viewBean->_view_body);
+			}
+		?>
+	</body>
 </html>
```
Comment (aethelwulffe) [inline interface/super/rules/base/template/basic.php]: Newline at end of file

File: library/translation.inc.php [hunk 4]
```diff
@@ -246,26 +243,6 @@ function getLanguageDir($lang_id) {
...
-HEADER HTML
-
-shows some informations for pages html header 
-
-@param none
-@return void
-*/
-function html_header_show() {
-    
-    // Below line was commented by the UTF-8 project on 05-2009 by BM.
```
Comment (aethelwulffe) [inline library/translation.inc.php]: And here was where the evil all started with forcing UTF-8

File: interface/patient_file/encounter/forms.php (hunk 2 above)
Comment (aethelwulffe) [reply interface/patient_file/encounter/forms.php]: Meaning:  opening tag was deleted.

**Comment by aethelwulffe** [comment]
Further testing:
echoed 
`<EMBED src="tada.wav" autostart=true loop=false volume=100 hidden=true><NOEMBED><BGSOUND src="tada.wav"></NOEMBED>`
into the function.  Tested.

Results:  Hilarious.

Switched to this branch.
Added the modified function back in.
No more cool noises.

Makes me wonder if we should start using audible errors...

Anyway @Jdew192837  please get the little changes in and I will proceed.

**Comment by aethelwulffe** [comment]
@Jdew192837 please look at the minor syntax errors above (closing tags especially) and get this fixed.  I am sick of seeing that function all over the place.  I want this merged.

**Comment by Jdew192837** [comment]
Okay

On Jan 26, 2018 12:56, "Art Eaton" <notifications@github.com> wrote:

@Jdew192837 <https://github.com/jdew192837> please look at the minor syntax
errors above (closing tags especially) and get this fixed. I am sick of
seeing that function all over the place. I want this merged.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/LibreHealthIO/LibreEHR/pull/938#issuecomment-360872620>,
or mute the thread

**Comment by Trodrige** [comment]
Hi, @Jdew192837 . We've moved to GitLab. PR owners are moving their work to GitLab. Can you do this buddy?

Thanks,
Rodrige.

**Comment by aethelwulffe** [comment]
We are just going to have to redo this work.  The last few changes for errors were not done, and it is way out of date.  It was a lot of work, but it will be easier to do over (and test) directory by directory.
